servers/physics_3d/godot_shape_3d.cpp

```diff
@@ -1473,35 +1473,56 @@ struct _Volume_BVH {
 	int face_index;
 };
 
-_Volume_BVH *_volume_build_bvh(_Volume_BVH_Element *p_elements, int p_size, int &count) {
-	_Volume_BVH *bvh = memnew(_Volume_BVH);
+struct _Volume_BVH_Pool {
```

Comment:
I think you could have used PagedAllocator which is basically a pool you can reuse.